scripts/builtin/statsNA.dml

```diff
@@ -0,0 +1,212 @@
+#-------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+
+# TODO Question: Is there a tool/template to generate the function documentation? Who updates the documentation on the website?
+
+m_statsNA = function(Matrix[Double] X, Integer bins = 4, Boolean print_result = TRUE)
+  return(Integer length_series,
+         Double number_nans,
+         Double percentage_nans,
+         Double number_nan_gaps,
+         Double average_size_nan_gaps,
+         Integer longest_nan_gap,
+         Integer most_frequent_nan_gap,
+         Integer most_weighty_nan_gap
+         )
+{
+    #TODO Question: Do we need to check if the input matrix is a vector? and if yes in with direction should it go?
+    if(ncol(X) != 1) {
+       stop("CONFUSION MATRIX: expect a matrix with only one column");
+    }
+
+    # Count total entries
+    # TODO Question: Why returns length a double and not an integer?
+    length_series = length(X);
+
+    # TODO Question: What should happen when the input matrix is empty?
+    if (length_series == 0) {
+       stop("EMPTY MATRIX")
+    }
+
+    # TODO Question: What should happen when the number of bins is higher than length of the input matrix?
+    if (length_series < bins) {
+        bins = length_series;
+    }
+
+    # Count NaNs
+    # TODO Question: Is there a style guide for private or return variables?
+    # TODO Question: Why didn't I find any documentation for "is.nan" ?
```

Comment:
New built-ins are added iteratively into the documentation.
is.na() or is.nan() will return a boolean mask matrix where the NaNs are represented as ones and other values as zeros.
 the following line of code could be written as,
p_position_nans = is.na(X)
p_position_nans = is.nan(X)
is.na() looks for both NA and NaN values while is.nan() specifically looks for NaNs.